velox/exec/RowContainer.h

```diff
@@ -1098,8 +1098,11 @@ class RowContainer {
       int32_t rightOffset,
       CompareFlags flags = CompareFlags());
 
-  // Free any variable-width fields associated with the 'rows'.
-  void freeVariableWidthFields(folly::Range<char**> rows);
+  // Free any variable-width fields associated with the 'rows'. If 'resetFields'
+  // is set to true, zero out complex-typed field in 'rows'.
+  void freeVariableWidthFields(
+      folly::Range<char**> rows,
+      bool resetFields = false);
```

Comment:
Any particular reason not to have zero out values all the time? Do we need this parameter? Why?

Reply:
Indeed this parameter is redundant.
My thought was it might be unnecessary to zero out when eraseRows. But since the rows will be added to free list and eventually given out via newRow, it's should be guaranteed that row return by newRow has no dirty data at complex type fields.
Fixed.